File: clients/java/examples/collections-count-all.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.collections.Collection;
import com.datastax.astra.client.collections.definition.documents.Document;
import com.datastax.astra.client.collections.exceptions.TooManyDocumentsToCountException;

public class Example {
  public static void main(String[] args) {
    // Get an existing collection
    Collection<Document> collection =
        new DataAPIClient("**APPLICATION_TOKEN**")
            .getDatabase("**API_ENDPOINT**")
            .getCollection("**COLLECTION_NAME**");

    // Count documents
    try {
      Integer result = collection.countDocuments(500);
      System.out.println(result);
    } catch (TooManyDocumentsToCountException error) {
      System.out.println("Number of documents exceeds upper bound or API limit");
    }
  }
}

File: clients/python/examples/collections-count-all.py
from astrapy import DataAPIClient, exceptions

# Get an existing collection
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)
collection = database.get_collection("**COLLECTION_NAME**")

# Count documents
try:
    result = collection.count_documents({}, upper_bound=500)
    print(result)
except exceptions.TooManyDocumentsToCountException:
    print("Number of documents exceeds upper bound or API limit")
